src/app/shared/carousel/carousel-directive.ts

```diff
@@ -0,0 +1,14 @@
+import {Directive, ElementRef, HostBinding} from '@angular/core';
+import {Carousel} from './carousel';
+
+@Directive({
+  selector: '[carousel-item]',
+})
+export class CarouselItemDirective {
+  @HostBinding('style.width') width = `${this.carousel.itemWidth}px`;
```

Comment:
Instead I think you can do `@HostBinding('style.width.px') width = this.carousel.itemWidth`